lib/index.d.ts

```diff
@@ -0,0 +1,54 @@
+interface Reviver {
+  (this: any, key: string, value: any): any;
+}
+
+interface Options {
+  /**
+   * - `'error'` - throw a `SyntaxError` when a `__proto__` key is found. This is the default value.
+   * - `'remove'` - deletes any `__proto__` keys from the result object.
+   * - `'ignore'` - skips all validation (same as calling `JSON.parse()` directly).
+   */
+  protoAction?: 'error' | 'remove' | 'ignore';
+}
+
+/**
+ * Parses a given JSON-formatted text into an object.
+ * @param text the JSON text string.
+ */
+export function parse(text: string): unknown;
+
+/**
+ * Parses a given JSON-formatted text into an object.
+ * @param text the JSON text string.
+ * @param reviver the `JSON.parse()` optional `reviver` argument.
+ */
+export function parse(text: string, reviver: Reviver): unknown;
+
+/**
+ * Parses a given JSON-formatted text into an object.
+ * @param text the JSON text string.
+ * @param options optional configuration object.
+ */
+export function parse(text: string, options: Options): unknown;
+
+/**
+ * Parses a given JSON-formatted text into an object.
+ * @param text the JSON text string.
+ * @param reviver the `JSON.parse()` optional `reviver` argument.
+ * @param options optional configuration object.
+ */
+export function parse(text: string, reviver: Reviver, options: Options): unknown;
+
+/**
+ * Scans a given object for prototype properties.
+ * @param obj the object being scanned.
+ * @param options optional configuration object.
+ */
+export function scan(obj: any, options?: Options): void;
```

Comment:
Maybe add a generic to allow you to specify the type of the scanned object?

```js
export function scan<T = any>(obj: T, options?: Options): void; 
```

Reply:
I'm not sure I understand what you expect from this change, how would people actually use this?

Reply:
It would allow people to explicitly specify the expected type, eg. `scan<MyObject>(…)`. It was just a thought, and probably not worth it to implement without a concrete use case.

Reply:
I can only think of developers mistakes for this, I don't think it's worth until we can figure out how it could be used.